fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/ExpressionOptimization.java

```diff
@@ -58,6 +59,20 @@ public class ExpressionOptimization extends ExpressionRewrite {
                     BetweenToEqual.INSTANCE
             )
     );
+
+    /**
+     * don't use it with PushDownFilterThroughJoin, it may cause dead loop:
+     *   LogicalFilter(origin expr)
+     *      => LogicalFilter((origin expr) and (add min max range))
+     *      => LogicalFilter((origin expr)) // use PushDownFilterThroughJoin
+     *      => ...
+     */
+    public static final List<ExpressionRewriteRule> ADD_RANGE = ImmutableList.of(
```

Comment:
pls add explain that why splitting this as a separate rule set

Reply:
is this possible to add other similar processing rule in the future?

Reply:
ming hong said it may add OrToIn to this rule set later